byterun/caml/mlvalues.h

```diff
@@ -186,7 +183,24 @@ bits  63        (64-P) (63-P)        10 9     8 7   0
 /* Pointer to the first field. */
 #define Op_val(x) ((value *) (x))
 /* Fields are numbered from 0. */
+#if CAML_API_VERSION < 405
 #define Field(x, i) (((value *)(x)) [i])           /* Also an l-value. */
+#endif
+
+static inline void caml_read_field (value obj, mlsize_t f, value* ret)
+{
+  *ret = Op_val(obj)[f];
+}
+
+static inline value Field_imm (value obj, mlsize_t f)
+{
+  return Op_val(obj)[f];
+}
+
+#define Int_field(x, i) Int_val(Op_val(x)[i])
+#define Long_field(x, i) Long_val(Op_val(x)[i])
```

Comment:
Hmm. We could remove `Int_val` and `Int_field` entirely (breaking even more code...), but I'm not sure what we could do short of that.